core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,553 @@
+    async fn block_number(&self) -> EnrichedClientResult<U64>;
+
+    /// Invokes a function on a contract specified by `contract_address` / `contract_abi` using `eth_call`.
+    async fn call_contract_function(
+        &self,
+        request: web3::CallRequest,
+        block: Option<BlockId>,
+    ) -> EnrichedClientResult<web3::Bytes>;
+}
+
+#[async_trait::async_trait]
+impl VerifierClient for PKSigningClient {
+    fn clone_boxed(&self) -> Box<dyn VerifierClient> {
+        Box::new(self.clone())
+    }
+
+    async fn block_number(&self) -> EnrichedClientResult<U64> {
+        self.as_ref().block_number().await
+    }
+
+    async fn call_contract_function(
+        &self,
+        request: web3::CallRequest,
+        block: Option<BlockId>,
+    ) -> EnrichedClientResult<web3::Bytes> {
+        self.as_ref().call_contract_function(request, block).await
+    }
+}
+
+#[derive(Debug)]
+pub enum VerificationError {
+    ServiceManagerError,
+    KzgError,
+    WrongProof,
+    DifferentCommitments,
+    DifferentRoots,
+    EmptyHash,
+    DifferentHashes,
+    WrongQuorumParams,
+    QuorumNotConfirmed,
+    CommitmentNotOnCurve,
+    CommitmentNotOnCorrectSubgroup,
+    LinkError,
+}
+
+/// Configuration for the verifier used for authenticated dispersals
+#[derive(Debug, Clone)]
+pub struct VerifierConfig {
+    pub rpc_url: String,
+    pub svc_manager_addr: String,
+    pub max_blob_size: u32,
+    pub g1_url: String,
+    pub g2_url: String,
+    pub settlement_layer_confirmation_depth: u32,
+    pub private_key: String,
+    pub chain_id: u64,
+}
+
+/// Verifier used to verify the integrity of the blob info
+/// Kzg is used for commitment verification
+/// EigenDA service manager is used to connect to the service manager contract
+#[derive(Debug)]
+pub struct Verifier {
+    kzg: Kzg,
+    cfg: VerifierConfig,
+    signing_client: Box<dyn VerifierClient>,
+}
+
+impl Clone for Verifier {
+    fn clone(&self) -> Self {
+        Self {
+            kzg: self.kzg.clone(),
+            cfg: self.cfg.clone(),
+            signing_client: self.signing_client.clone_boxed(),
+        }
+    }
+}
+
+impl Verifier {
+    pub const DEFAULT_PRIORITY_FEE_PER_GAS: u64 = 100;
+    pub const SRSORDER: u32 = 268435456; // 2 ^ 28
+    pub const G1POINT: &'static str = "g1.point";
+    pub const G2POINT: &'static str = "g2.point.powerOf2";
+    pub const POINT_SIZE: u32 = 32;
+
+    async fn save_point(url: String, point: String) -> Result<(), VerificationError> {
+        let url = Url::parse(&url).map_err(|_| VerificationError::LinkError)?;
+        let response = reqwest::get(url)
+            .await
+            .map_err(|_| VerificationError::LinkError)?;
+        if !response.status().is_success() {
+            return Err(VerificationError::LinkError);
+        }
+        let path = format!("./{}", point);
+        let path = Path::new(&path);
+        let mut file = File::create(path).map_err(|_| VerificationError::LinkError)?;
```

Comment:
I think we can leave it as a future improvement since downloading it is the way it is handled in other projects.
Also these points are downloaded only 1 time when the node starts. So network errors are not a problem here.

Having it inside a docker image sounds good but it requires further coordination I guess since it implies modifying the zksync-era images. 

> Please use Tokio APIs for managing files here; blocking ones from std do not interact well with the Tokio runtime.

Fixed here https://github.com/lambdaclass/zksync-era/pull/375